gnpy/core/elements.py

```diff
@@ -624,14 +624,8 @@ def _nf(self, type_def, nf_model, nf_fit_coeff, gain_min, gain_flatmax, gain_tar
             nf_avg = pin_ch - polyval(nf_model.nf_coef, pin_ch) + 58
         elif type_def == 'advanced_model':
             nf_avg = polyval(nf_fit_coeff, -dg)
-        else :
-            print(
-                f'\x1b[1;31;40m'\
-                + f'CRITICAL: unrecognized type def _{self.params.type_def}_\n\
-                    => please check eqpt_config.json'\
-                + '\x1b[0m'
-                )                        
-            exit()            
+        else:
+            assert False, "Unrecognized amplifier type, this should have been checked by the JSON loader"
```

Comment:
It might be usefull to show which  type is invalid here , I think it is self.params.type_def

Reply:
Thanks for checking this patch series, Esther.

I've used an assertion instead of a regular exception here because this code path will not be ever taken unless our code is buggy. This is "risky" because assertions are turned into a no-op when optimizations are active, so I would prefer not to add more information here -- they might not even be shown.